examples/using-i18n/src/pages/index.js

```diff
@@ -0,0 +1,56 @@
+import React from "react"
+import { graphql } from "gatsby"
+import LocalizedLink from "../components/localizedLink"
+import useTranslations from "../components/useTranslations"
+
+const Index = ({ data: { allMdx } }) => {
+  // useTranslations is aware of the global context (and therefore also "locale")
+  // so it'll automatically give back the right translations
+  const { hello, subline } = useTranslations()
+
+  return (
+    <>
+      <h1>{hello}</h1>
+      <p>{subline}</p>
+      <hr style={{ margin: `2rem 0` }} />
+      <ul className="post-list">
+        {allMdx.edges.map(({ node: post }) => (
+          <li key={`${post.frontmatter.title}-${post.fields.locale}`}>
+            <LocalizedLink to={`/${post.parent.relativeDirectory}`}>
+              {post.frontmatter.title}
+            </LocalizedLink>
+            <div>{post.frontmatter.date}</div>
+          </li>
+        ))}
+      </ul>
+    </>
+  )
+}
+
+export default Index
+
+export const query = graphql`
+  query Index($locale: String!) {
+    allMdx(
+      filter: { fields: { locale: { eq: $locale } } }
+      sort: { fields: [frontmatter___date], order: DESC }
+    ) {
+      edges {
+        node {
+          frontmatter {
+            title
+            date(formatString: "DD/MM/YYYY")
```

Comment:
date should be also i18n for example:
- german: DD.MM.YYYY
- english: MM/DD/YYYY

Reply:
> english: MM/DD/YYYY

That format doesn't apply to all countries where English is the dominant language. Here in New Zealand it's commonly: `DD/MM/YYYY`.

This is more l10n (localisation) specific. You can see a good example of this on the [moment.js homepage](http://momentjs.com/), scroll to the bottom and see the different english locales. They also demonstrate ways to format the date to what you'd like while respecting the locale.

Alternatively something like [`toLocaleDateString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) might also work? It's a lightweight/native alternative to something like Moment, which follows the official ECMAScript Internationalization API spec.

Reply:
Yeah, I don't really want to get into localization. But I passed the dateFormat as context, too. So both pages now show a different format :)